File: lab1/lab1.cpp
#define _USE_MATH_DEFINES

#include <cmath>
#include <iostream>


int main()
{
    float triangleSide1, triangleSide2, triangleAngle, temp = 0;
    std::string flag = "";
    while (true) {
        std::cout << "Enter first side:";
        std::cin >> triangleSide1;
        std::cout << "Enter second side:";
        std::cin >> triangleSide2;
        std::cout << "Enter angle:";
        std::cin >> temp; 
        triangleAngle = (temp / 180 )* M_PI;

        if (triangleAngle == 0 || triangleSide1 == 0 || triangleSide2 == 0) {
            std::cout << "Error (incorrect data). Try again";
            continue;
        }
        std::cout  << 0.5 * triangleSide1 * triangleSide2 * sin(triangleAngle) << '\n';
        std::cout << "Countinue ?(y/n): ";
        std::cin >> flag;
        if (flag == "n") {
            break;
        }
    }
}

File: lab1.py
import math
triangleAngle = 0
triangleSide1 = 0
triangleSide2 = 0 

while(True):
    triangleSide1 = float(input('Enter first side:'))
    triangleSide2 = float(input('Enter second side:'))
    triangleAngle = float(input('Enter angle:')) /180 * math.pi
    if (triangleAngle == 0 or triangleSide1 == 0 or triangleSide2 == 0):
        print('Error(incorrect data).Try again')
        continue 
    print(0.5 * triangleSide1 * triangleSide2 * math.sin(triangleAngle)) 
    if (input('Continue?(y/n) :') == 'n'):
        break